clang/docs/checkedc/Update-to-latest-LLVM-sources.md

```diff
@@ -1,81 +1,212 @@
+guidelines on fixing the merge conflicts and test failures.
+```
+    git checkout updated_baseline_master_12
+    git merge master > merge_conflict_list.txt
+``` 
+7. The resolution of test failures may require you to cherry-pick commits from
+the upstream LLVM/Clang repository. Record the cherry-picked commits in the
+document LLVM-Upgrade-Notes.md. Let &lt;commit-to-cherry-pick&gt; be one
+such commit hash on branch &lt;cherry-pick-branch&gt; of the LLVM/Clang
+repository. 
+```
+    git fetch upstream <cherry-pick-branch>
+    git cherry-pick <commit-to-cherry-pick>
+```
+8. While the merge conflicts and test failures are being fixed, it is possible
+that the `master` branch has had several commits. Merge the `master` branch into
+`updated_baseline_master_12` branch, and resolve the new conflicts and failures
+if necessary.
+```
+    git merge master >> merge_conflicts_list.txt
+```
+9. After all the merge conflicts and test failures on the local machine are
+fixed, push the `baseline` and `updated_baseline_master_12` branches to the
+remote repository and excute the ADO tests on `updated_baseline_master_12`.
+```
+    git push origin baseline
+    git push origin updated_baseline_master_12
+```
+10. After the ADO tests are successful, merge the `updated_baseline_master_12`
+branch into `master` and push to the remote repository.
+```
+    git checkout master
+    git merge updated_baseline_master_12
+    git push origin master
+```
+
+## Detailed Instructions - Phase 2
+We give the instructions for the first sub-phase of Phase 2. Similar steps
+have to be followed for the other sub-phase.
+
+1. Create a new branch, called `release_12.x`, from the `master` branch in the
+`checkedc-clang` repository. If the branch already exists, then merge the
+`master` branch into it. Resolve merge conflicts and test failures if any.
+```        
+    git checkout -b release_12.x master
+    OR
+    git checkout release_12.x
     git merge master
-
-You will very likely have merge conflicts and some test failures.  The test
-failures usually stem from incorrect merges or Checked C-specific data not being
-initialized by new or changed constructor methods.
-
-You may also need to pick up changes from LLVM/Clang for fixes to any unexpected
-baseline failures.
-
-You can push your updated master branches up to GitHub for automated
-testing.  If you haven't cleaned the build directory as described earlier,
-make sure you do that.
-
-You'll want to run automated tests on Linux and Windows x86/x64, as well as
+```     
+2. Upgrade the branch `release_12.x` to the LLVM/Clang sources corresponding to
+the 12.0.0 release. For this, we need to get the commit hash associated with the
+12.0.0 release from https://github.com/llvm/llvm-project/releases. Let this 
+commit hash be &lt;12.0.0-commit-hash&gt;. Note that it should be the full
+commit hash. Resolve merge conflicts and test failures if any.
```

Comment:
```suggestion
the 12.0.0 release, given by the `llvmorg-12.0.0` tag. Resolve
merge conflicts and test failures if any.
```
Another case where you can use a Git ref provided by LLVM rather than manually looking up a commit ID.